src/generators/cplusplus/CplusplusConstrainer.ts

```diff
@@ -0,0 +1,71 @@
+import { TypeMapping } from '../../helpers';
+import {
+  defaultEnumKeyConstraints,
+  defaultEnumValueConstraints
+} from './constrainer/EnumConstrainer';
+import { defaultModelNameConstraints } from './constrainer/ModelNameConstrainer';
+import { defaultPropertyKeyConstraints } from './constrainer/PropertyKeyConstrainer';
+import { CplusplusDependencyManager } from './CplusplusDependencyManager';
+import { CplusplusOptions } from './CplusplusGenerator';
+
+export const CplusplusDefaultTypeMapping: TypeMapping<
+  CplusplusOptions,
+  CplusplusDependencyManager
+> = {
+  Object({ constrainedModel }): string {
+    //Returning name here because all object models have been split out
+    return `${constrainedModel.name}*`;
+  },
+  Reference({ constrainedModel }): string {
+    return `${constrainedModel.name}*`;
```

Comment:
Here there is the additional question of whether or not the reference allows the user to modify the referenced value. If modifications are not intended, we should add the `const` keyword to before the `*`.

Reply:
I dont see how we can force no modifications here 🤔 

We could potentially add this as an option later down the line if it gets requested?

Reply:
If the output were `${constrainedModel.name} const*`, that would mean that whoever is accessing this property, cannot modify it. My question was more about the theoretical model than the C++ output though, how these References are intended to behave.

Reply:
Is that not up to the individual use-case, i.e. not really something we can hardcode but should be conditional based on what the user want?

Reply:
Now I finally understand this, and the desired solution is to have only `${constrainedModel.name}`, without the `*`. I was initially confused by the naming, but here we are creating a reference to a type, not to an object. It is important to note that the referenced type must be defined before the current type is defined.